[quarkus-next] [windows] ClassNotFoundException: JvmOptionsBuilder

### Before reporting an issue

- [x] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

dist/quarkus

### Describe the bug

Matrix: `Quarkus UT / windows-latest`
Job: https://github.com/keycloak/keycloak/actions/runs/16557076973/job/46820232493

Error cause: `java.lang.ClassNotFoundException: io.quarkus.bootstrap.model.JvmOptionsBuilder`

Suspecting this issue as the cause of these test failures: https://github.com/keycloak/keycloak/issues/41466

Full log:
```shell
[INFO] Reactor Summary for Keycloak Quarkus Parent 999.0.0-SNAPSHOT:
[INFO] 
[INFO] Keycloak Quarkus Parent ............................ SUCCESS [  1.629 s]
[INFO] Keycloak Configuration API ......................... SUCCESS [  5.527 s]
[INFO] Keycloak Quarkus Server Extension .................. SUCCESS [ 23.746 s]
[INFO] Keycloak Quarkus Server Deployment ................. FAILURE [02:37 min]
[INFO] Keycloak Quarkus Server App ........................ SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  03:08 min
[INFO] Finished at: 2025-07-28T00:27:58Z
[INFO] ------------------------------------------------------------------------
Error:  Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:3.0.0-M7:test (default-test) on project keycloak-quarkus-server-deployment: 
Error:  
Error:  Please refer to D:\a\keycloak\keycloak\quarkus\deployment\target\surefire-reports for the individual test results.
Error:  Please refer to dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.
Error: [ERROR] There was an error in the forked process
Error:  TestEngine with ID 'junit-jupiter' encountered a critical issue during test discovery:
Error:  
Error: RROR] ClassSelector [className = 'test.org.keycloak.quarkus.services.health.KeycloakMetricsConfigurationTest', classLoader = null] resolution failed
Error:      Source: ClassSource [className = 'test.org.keycloak.quarkus.services.health.KeycloakMetricsConfigurationTest', filePosition = null]
Error:              at test.org.keycloak.quarkus.services.health.KeycloakMetricsConfigurationTest.<no-method>(SourceFile:0)
Error:      Cause: java.lang.NoClassDefFoundError: io/quarkus/bootstrap/model/JvmOptionsBuilder
Error:  	at io.quarkus.bootstrap.model.ApplicationModelBuilder.handleExtensionProperties(ApplicationModelBuilder.java:188)
Error:  	at io.quarkus.bootstrap.resolver.maven.ExtensionInfo.ensureActivated(ExtensionInfo.java:54)
Error:  	at io.quarkus.bootstrap.resolver.maven.ApplicationDependencyResolver$AppDep.setFlags(ApplicationDependencyResolver.java:640)
Error:  	at io.quarkus.bootstrap.resolver.maven.ApplicationDependencyResolver$AppDep.setChildFlags(ApplicationDependencyResolver.java:620)
Error:  	at io.quarkus.bootstrap.resolver.maven.ApplicationDependencyResolver.processRuntimeDeps(ApplicationDependencyResolver.java:466)
Error:  	at io.quarkus.bootstrap.resolver.maven.ApplicationDependencyResolver.resolve(ApplicationDependencyResolver.java:202)
Error:  	at io.quarkus.bootstrap.resolver.BootstrapAppModelResolver.buildAppModel(BootstrapAppModelResolver.java:424)
Error:  	at io.quarkus.bootstrap.resolver.BootstrapAppModelResolver.doResolveModel(BootstrapAppModelResolver.java:357)
Error:  	at io.quarkus.bootstrap.resolver.BootstrapAppModelResolver.resolveManagedModel(BootstrapAppModelResolver.java:238)
Error:  	at io.quarkus.bootstrap.BootstrapAppModelFactory.resolveAppModelForWorkspace(BootstrapAppModelFactory.java:266)
Error:  	at io.quarkus.bootstrap.BootstrapAppModelFactory.resolveAppModel(BootstrapAppModelFactory.java:216)
Error:  	at io.quarkus.test.junit.AppMakerHelper.makeCuratedApplication(AppMakerHelper.java:208)
Error:  	at io.quarkus.test.junit.classloading.FacadeClassLoader.getOrCreateCuratedApplication(FacadeClassLoader.java:489)
Error:  	at io.quarkus.test.junit.classloading.FacadeClassLoader.getOrCreateRuntimeClassLoader(FacadeClassLoader.java:508)
Error:  	at io.quarkus.test.junit.classloading.FacadeClassLoader.getQuarkusClassLoader(FacadeClassLoader.java:414)
Error:  	at io.quarkus.test.junit.classloading.FacadeClassLoader.loadClass(FacadeClassLoader.java:326)
Error:  	at java.base/java.lang.Class.forName0(Native Method)
Error:  	at java.base/java.lang.Class.forName(Class.java:534)
Error:  	at java.base/java.lang.Class.forName(Class.java:513)
Error:  	at org.junit.platform.commons.util.ReflectionUtils.lambda$tryToLoadClass$10(ReflectionUtils.java:886)
Error:  	at org.junit.platform.commons.function.Try.lambda$call$0(Try.java:57)
Error:  	at org.junit.platform.commons.function.Try.of(Try.java:93)
Error:  	at org.junit.platform.commons.function.Try.call(Try.java:57)
Error:  	at org.junit.platform.commons.util.ReflectionUtils.tryToLoadClass(ReflectionUtils.java:873)
Error:  	at org.junit.platform.commons.util.ReflectionUtils.tryToLoadClass(ReflectionUtils.java:810)
Error:  	at org.junit.platform.commons.support.ReflectionSupport.tryToLoadClass(ReflectionSupport.java:94)
Error:  	at org.junit.platform.engine.discovery.ClassSelector.getJavaClass(ClassSelector.java:95)
Error:  	at org.junit.jupiter.engine.discovery.ClassSelectorResolver.resolve(ClassSelectorResolver.java:87)
Error:  	at org.junit.platform.engine.support.discovery.EngineDiscoveryRequestResolution.lambda$resolve$2(EngineDiscoveryRequestResolution.java:135)
Error:  	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
Error:  	at java.base/java.util.ArrayList$ArrayListSpliterator.tryAdvance(ArrayList.java:1685)
Error:  	at java.base/java.util.stream.ReferencePipeline.forEachWithCancel(ReferencePipeline.java:129)
Error:  	at java.base/java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:527)
Error:  	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:513)
Error:  	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
Error:  	at java.base/java.util.stream.FindOps$FindOp.evaluateSequential(FindOps.java:150)
Error:  	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
Error:  	at java.base/java.util.stream.ReferencePipeline.findFirst(ReferencePipeline.java:647)
Error:  	at org.junit.platform.engine.support.discovery.EngineDiscoveryRequestResolution.resolve(EngineDiscoveryRequestResolution.java:189)
Error:  	at org.junit.platform.engine.support.discovery.EngineDiscoveryRequestResolution.resolve(EngineDiscoveryRequestResolution.java:126)
Error:  	at org.junit.platform.engine.support.discovery.EngineDiscoveryRequestResolution.resolveCompletely(EngineDiscoveryRequestResolution.java:92)
Error:  	at org.junit.platform.engine.support.discovery.EngineDiscoveryRequestResolution.run(EngineDiscoveryRequestResolution.java:83)
Error:  	at org.junit.platform.engine.support.discovery.EngineDiscoveryRequestResolver.resolve(EngineDiscoveryRequestResolver.java:148)
Error:  	at org.junit.jupiter.engine.discovery.DiscoverySelectorResolver.resolveSelectors(DiscoverySelectorResolver.java:63)
Error:  	at org.junit.jupiter.engine.JupiterTestEngine.discover(JupiterTestEngine.java:70)
Error:  	at org.junit.platform.launcher.core.EngineDiscoveryOrchestrator.discoverEngineRoot(EngineDiscoveryOrchestrator.java:195)
Error:  	at org.junit.platform.launcher.core.EngineDiscoveryOrchestrator.discoverSafely(EngineDiscoveryOrchestrator.java:174)
Error:  	at org.junit.platform.launcher.core.EngineDiscoveryOrchestrator.discover(EngineDiscoveryOrchestrator.java:119)
Error:  	at org.junit.platform.launcher.core.EngineDiscoveryOrchestrator.discover(EngineDiscoveryOrchestrator.java:84)
Error:  	at org.junit.platform.launcher.core.DefaultLauncher.discover(DefaultLauncher.java:104)
Error:  	at org.junit.platform.launcher.core.DefaultLauncher.discover(DefaultLauncher.java:83)
Error:  	at org.junit.platform.launcher.core.DelegatingLauncher.discover(DelegatingLauncher.java:42)
Error:  	at org.junit.platform.launcher.core.InterceptingLauncher.lambda$discover$0(InterceptingLauncher.java:33)
Error:  	at org.junit.platform.launcher.core.ClasspathAlignmentCheckingLauncherInterceptor.intercept(ClasspathAlignmentCheckingLauncherInterceptor.java:25)
Error:  	at org.junit.platform.launcher.core.InterceptingLauncher.discover(InterceptingLauncher.java:33)
Error:  	at org.junit.platform.launcher.core.DelegatingLauncher.discover(DelegatingLauncher.java:42)
Error:  	at org.apache.maven.surefire.junitplatform.LazyLauncher.discover(LazyLauncher.java:48)
Error:  	at org.apache.maven.surefire.junitplatform.TestPlanScannerFilter.accept(TestPlanScannerFilter.java:56)
Error:  	at org.apache.maven.surefire.api.util.DefaultScanResult.applyFilter(DefaultScanResult.java:102)
Error:  	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.scanClasspath(JUnitPlatformProvider.java:167)
Error:  	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invoke(JUnitPlatformProvider.java:139)
Error:  	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:456)
Error:  	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:169)
Error:  	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:595)
Error:  	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:581)
Error:  Caused by: java.lang.ClassNotFoundException: io.quarkus.bootstrap.model.JvmOptionsBuilder
Error:  	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641)
Error:  	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188)
Error:  	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:526)
Error:  	... 65 more
```

### Version

quarkus-next

### Regression

- [ ] The issue is a regression

### Expected behavior

Tests pass

### Actual behavior

Tests fail

### How to Reproduce?

Look at the `quarkus-next` runs

### Anything else?

_No response_

--- a/quarkus/deployment/pom.xml
+++ b/quarkus/deployment/pom.xml
@@ -298,6 +298,8 @@
                 <configuration>
                     <systemPropertyVariables>
                         <java.util.logging.manager>org.jboss.logmanager.LogManager</java.util.logging.manager>
+                        <!-- We want to avoid any other artifacts fetches done via surefire and @QuarkusTest - see https://github.com/keycloak/keycloak/issues/41466 -->
+                        <maven.settings>void</maven.settings>
                     </systemPropertyVariables>
                 </configuration>
             </plugin>

